rust/src/operations/write.rs

```diff
@@ -1,517 +1,508 @@
+                        if batches.is_empty() {
+                            return Err(WriteError::MissingData.into());
+                        }
+                        Ok(batches[0].schema().try_into()?)
+                    } else {
+                        Err(WriteError::MissingData)
+                    }?;
+                    let mut builder = CreateBuilder::new()
+                        .with_object_store(table.object_store())
+                        .with_columns(schema.get_fields().clone());
+                    if let Some(partition_columns) = this.partition_columns.as_ref() {
+                        builder = builder.with_partition_columns(partition_columns.clone())
                     }
-                    SaveMode::Ignore => {
-                        return Ok(Box::pin(SizedRecordBatchStream::new(
-                            schema,
-                            vec![],
-                            tracking_metrics,
-                        )))
+                    let (_, actions, _) = builder.into_table_and_actions()?;
+                    Ok(actions)
+                }
+                Ok(_) => {
+                    if table.get_min_writer_version() > MAX_SUPPORTED_WRITER_VERSION {
+                        Err(
+                            WriteError::UnsupportedWriterVersion(table.get_min_writer_version())
+                                .into(),
+                        )
+                    } else {
+                        match this.mode {
+                            SaveMode::ErrorIfExists => {
+                                Err(WriteError::AlreadyExists(table.table_uri()).into())
+                            }
+                            _ => Ok(vec![]),
+                        }
                     }
-                    _ => Ok(vec![]),
                 }
+                Err(err) => Err(err),
+            }?;
+
+            // validate partition columns
+            let partition_columns = if let Ok(meta) = table.get_metadata() {
+                if let Some(ref partition_columns) = this.partition_columns {
+                    if &meta.partition_columns != partition_columns {
+                        Err(WriteError::PartitionColumnMismatch {
+                            expected: table.get_metadata()?.partition_columns.clone(),
+                            got: partition_columns.to_vec(),
+                        })
+                    } else {
+                        Ok(partition_columns.clone())
+                    }
+                } else {
+                    Ok(meta.partition_columns.clone())
+                }
+            } else {
+                Ok(this.partition_columns.unwrap_or_default())
+            }?;
+
+            let plan = if let Some(plan) = this.input {
+                Ok(plan)
+            } else if let Some(batches) = this.batches {
+                if batches.is_empty() {
+                    Err(WriteError::MissingData)
```

Comment:
I guess ... :). if we do we need to either make it a no-op or add some logic to handle the case when we do not have an explicitly defined table schema, in case we need to create the table.